torch/csrc/distributed/c10d/reducer.cpp

```diff
@@ -730,6 +753,59 @@ void Reducer::finalize_backward() {
     local_used_work_->wait();
   }
   local_used_maps_reduced_ = false;
+
+  // Rebuilt bucket if this is the first time to rebuilt
+  if (!rebuilt_tensors_.empty()) {
+    rebuildBuckets();
+  }
+}
+
+void Reducer::rebuildBuckets() {
+  TORCH_INTERNAL_ASSERT(
+      rebuilt_tensors_.size() == rebuilt_param_indices_.size(),
+      "rebuild tensor size is not same as rebuild param indices size.");
+  std::vector<std::vector<size_t>> rebuilt_bucket_indices;
+  std::vector<size_t> bucket_size_limits;
+  bucket_size_limits.push_back(DEFAULT_FIRST_BUCKET_BYTES);
+  bucket_size_limits.push_back(bucket_bytes_cap_);
+  rebuilt_bucket_indices = compute_bucket_assignment_by_size(
+      rebuilt_tensors_,
```

Comment:
nit discussion: is it intentional for using `rebuilt_tensors_` instead of `rebuild_tensors_`?

Reply:
feel rebuild_tensors_ means an action, but rebuilt_tensors_ means a noun, so using rebuilt_... for those variables, how do you think?